qutebrowser/config/configdata.yml

```diff
@@ -561,22 +570,32 @@ content.host_blocking.lists:
 
     The file `~/.config/qutebrowser/blocked-hosts` is always read if it exists.
 
-content.host_blocking.whitelist:
+content.blocking.adblock.enabled:
```

Comment:
I thought that giving people the choice to use both would be good, as some might want to. But perhaps it is simpler to allow only one method at a time? Should we gather community opinion?

> How should we handle the case where this is true (the default), but the library isn't available?
```python
@cmdutils.register()
def brave_adblock_update() -> None:
    """Update the adblock block lists."""

    if _ad_blocker is not None:
        _ad_blocker.adblock_update()
    else:
        message.warning(
            "The 'adblock' dependency is not installed. Please install it to continue."
        )
```
Currently, when the `:brave-adblock-update` command is run and the `adblock` dependency is not present, a warning is displayed. No ads are blocked. Maybe there is a more user friendly alternative where we fall back on host blocking?

Reply:
I would personally use both methods. That was my 2 cents.

Reply:
I'd definitely like to fall back on host blocking by default rather than having no blocking at all. See my proposal in https://github.com/qutebrowser/qutebrowser/pull/5317#issuecomment-624131110 where I suggested to have an `auto` default, and also a `both` option which then enables both. The main benefit I see over two separate `enabled` options is that we can default to `auto` rather than having to decide which blocking method is enabled by default.

Reply:
Can I suggest an option which combines both perspectives?

I think `content.blocking.hosts.enabled` should take the values `true` or `false` (default is up to you), and `conent.blocking.adblock.enabled` should take the values `true`, `false`, or `auto` (default, only if dependency exists).

I'm not sure if qutebrowser's config system lets you have options with two different types though.

Reply:
Can I suggest an option which combines both perspectives?

I think `content.blocking.hosts.enabled` should take the values `true` or `false` (default is up to you), and `conent.blocking.adblock.enabled` should take the values `true`, `false`, or `auto` (default, only if dependency exists).

I'm not sure if qutebrowser's config system lets you have options with two different types though.

Reply:
@arjvik That would work (with a custom `String` type), but I'm not convinced it is a good idea - are there any arguments for that over a `content.blocking.method` that I'm missing? The main argument *against* it that I'm seeing is the default for `content.blocking.hosts.enabled`:

- If it's disabled by default: There's no adblocking at all without the `adblock` dependency by default, which I'd rather like to avoid: Even now with an `:adblock-update` being needed before ads are blocked (and with a persistent statusbar notification about that on every start) there are already a lot of people claiming that qutebrowser doesn't have an adblocker at all.
- If it's enabled by default: There are two adblockers running if the `adblock` dependency is available by default. Thus, there's more stuff running on every request (which is somewhat performance-sensitive) - and if someone wants to disable blocking for a page, they need to disable two different blockers. I think there are just more problems than it's worth from a user perspective...

@ArniDagur What are your views on this?

Reply:
After some thinking, I think I agree with https://github.com/qutebrowser/qutebrowser/pull/5317#issuecomment-624131110 which says:
```
I think this should be a separate content.blocking.method setting, with values like the following:

    auto: Use brave adblocking if available, host blocking otherwise (default)
    adblock: Use brave adblocking (show a warning if unavailable)
    hosts: Use hosts adblocking
    both: Use both (show a warning if the adblock module is unavailable)
```
It's worth noting that upstream recently added support (https://github.com/brave/adblock-rust/commit/ff3daf6d5774be7f85f98da85ae42d40fb744b61) for host blocking as well, so perhaps under the `both`  setting we could use that?

Reply:
I don't think it makes sense to use the adblock-rust library for host blocking - the implementation in qutebrowser is quite straightforward and we need to keep it around anyways. I think I'd rather keep using that, to also make debugging and such easier in case of issues.